src/lib/releaseTestPlanGenerator.ts

```diff
@@ -0,0 +1,171 @@
+      files.map(filename => loadTestPlanFromFile(filename))
+   );
+
+   // Multiple test plans might require the same test case
+   // which requires merging to consolidate their required status
+   const mergedTestCases: TestCase[] = [];
+
+   testPlans
+      .filter(testPlan => !!testPlan.testCases)
+      .map(testPlan => {
+         testPlan.testCases.map(testCase =>
+            mergeTestCase(
+               testCase,
+               getFilenameFromCsv(testPlan.name),
+               mergedTestCases
+            )
+         );
+      });
+
+   const mergedTestPlan: TestPlan = {
+      testCases: mergedTestCases,
+      name: filename,
+      versionToTest: versionToTest,
+   };
+   return await exportTestPlan(mergedTestPlan);
+}
+
+/**
+ * Merge a Test Case when it has already been required by another Test Plan
+ */
+export function mergeTestCase(
+   testCase: TestCase,
+   requiredByTestPlanName: string,
+   mergedTestCases: TestCase[]
+): void {
+   // Test case does not exists, add it
+   if (!testCaseExists(testCase, mergedTestCases)) {
+      var newRequiredBy = testCase.isRequired ? requiredByTestPlanName : "";
+
+      mergedTestCases.push({
+         featureName: testCase.featureName,
+         isRequired: testCase.isRequired,
+         requiredBy: newRequiredBy,
+      });
+   } else {
+      // Test case exists, requires merging
+      const existingTestCaseIndex = findIndexForTestCase(
+         testCase,
+         mergedTestCases
+      );
+
+      if (testCase.isRequired) {
+         if (mergedTestCases[existingTestCaseIndex].isRequired) {
+            // Existing test case was required by another test plan, augment requiredBy with this one
+            mergedTestCases[existingTestCaseIndex].requiredBy =
+               mergedTestCases[existingTestCaseIndex].requiredBy +
+               " " +
+               requiredByTestPlanName;
+         } else {
+            // Existing test case was not previously required, add only this requireBy
+            mergedTestCases[
+               existingTestCaseIndex
+            ].requiredBy = requiredByTestPlanName;
+         }
+
+         // Ensure the test case is now required
+         mergedTestCases[existingTestCaseIndex].isRequired = true;
+      }
+   }
+}
+
+/**
+ * Load the content of a Test Plan that was previously saved to cvs file
+ * @param filename
+ */
+export async function loadTestPlanFromFile(
+   filename: string
+): Promise<TestPlan> {
+   const data = await readContentFromFile(filename);
+
+   if (!data) {
+      new Error(`Cannot read test plan: ${filename}`);
+   }
+
+   const version = loadVersionToTest(data);
+
+   if (!version) {
+      new Error(`Cannot read Version to Test in ${filename}`);
+   }
+
+   const testPlanScenarios = await loadTestCases(data);
+
+   return {
+      versionToTest: version,
+      testCases: testPlanScenarios,
+      name: filename,
+   };
+}
+
+export function loadVersionToTest(data: string) {
```

Comment:
You might consider naming these next two `parse*`, I think it has slightly better semantics

Reply:
So true!